compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/PersistRecordValidator.java

```diff
@@ -285,9 +327,10 @@ private void checkAutoIncrementFieldMarkAsKey(SyntaxNodeAnalysisContext ctx, Loc
     }
 
     private void validateEntityAnnotation(SyntaxNodeAnalysisContext ctx, TypeDefinitionNode typeDefinitionNode,
-                                          RecordTypeSymbol recordTypeSymbol) {
+                                          RecordTypeSymbol recordTypeSymbol, DocumentId documentId) {
```

Comment:
documentId is passed only to get the path. Cant we process and pass it down? Again whats the use of passing ctx to validate method? Unless there is a need to process it except for diagnostic reporting, it should not be passed down